src/runner.ts

```diff
@@ -241,19 +250,23 @@ export class ResolveRunner implements Types.IResolveRunner {
     let ref = new URI(refStr);
 
     // Does ref only have a fragment
-    if (ref.toString() !== `#${ref.fragment()}`) {
-      let isFile = ref.scheme() === 'file';
-
-      // if the file scheme is not explicitly set, check the authority
-      if (!isFile && !ref.scheme()) {
-        const scheme = this.authority.scheme();
-        // if the authority has no scheme, then assume it is a file (urls will have http, etc)
-        isFile = scheme === '' || scheme === 'file';
-      }
+    if (ref.toString().charAt(0) !== '#') {
+      const isFile = this.isFile(ref);
 
       // if we're working with a file, resolve any path diferences and make sure the scheme is set
       if (isFile) {
-        ref = ref.absoluteTo(this.authority).scheme('file');
+        let absRef = ref.toString();
+        if (!ref.is('absolute')) {
+          if (this.authority.toString()) {
+            absRef = join(dirname(this.authority.toString()), absRef);
```

Comment:
I'm kinda hesitant to use `path.join` and `path.dirname` on URIs - in this case it works because they are relative and we know it's a file... but I'm wonder if there's an edge case we might not be aware of.

Reply:
Hmm, I can't think of any - I added some test cases to try to cover the edge cases I could think of though.

Reply:
I posted an alternative to `dirname` in Slack. I'm actually helping paint the house this weekend, but... these are the things I end up thinking about. :)

Reply:
Or... shouldn't we just use URI.js here to compute the absolute path?

Reply:
Does not work under Windows. See Studio branch and the comment I left.